test/juxt/site/graphql_test.clj

```diff
@@ -301,6 +301,164 @@ type Mutation {
 (deftest mutation-test
   (mutation))
 
+(defn mutation-with-validation-directive []
+  (let [schema "
+schema { query: Query mutation: Mutation }
+type Query { person: Person }
+type Person { id: ID @site(a: \"xt/id\") name: String }
+input PersonInput { name: String! id: ID }
+
+type Mutation {
+  addPerson(
+    person: PersonInput
+  ): Person @site(
+    validation: {
+      person: \"[:map [:id {:optional true} :any] [:name [:string {:min 1 :max 20}]]]\"
+    }
+  )
+}"
+
+        query "
+mutation { addPerson(
+  person: {id: \"https://example.org/persons/mal\" name: \"Malcolm Sparks\"}) { id name }}
+"]
```

Comment:
I think its probably easier to read if this schema and query are def'd outside the test, or maybe even in .graphql files and slurped in so they are easier to edit. In the file has the advantage of colocation though